### DISCUSSION TITLE
Template Regex Constraints

### DISCUSSION THREAD

**Discussion by khepin** [cuelang/cue#659] [Q&A]
When researching an answer to https://github.com/cuelang/cue/discussions/657 (restrict string length), I had the following regex:

```cue
#StrLen8: =~"^.{8}$"
{
    v: #StrLen8
}
```
Which works, but only for length 8. So, loosely inspired by the documentation at https://cuelang.org/docs/tutorials/tour/references/emit/ and https://cuelang.org/docs/tutorials/tour/types/templates/ I thought I could apply that to regexes and attempted:
```cue
#StrLen: [len=_]: =~"^.{\(len)}$"
{
    v: #StrLen : 8
}
```
... that does NOT work ;-) is there a way to template regexes like this?

**Comment by mpvl** [comment]
See my answer in #657.

In this particular case, pattern constraints cannot match against numbers and must match against fields.

**Reply by khepin** [reply]
So the suggestion in that answer is:

```
#StrLen: {
	=~"^.{\(#n)}$"
	#n: int
}
#StrLen8: { #StrLen, #n: 8 }
```
Which actually gives me an error: 

```
#StrLen: can only embed structs (found string):
    ./test.cue:2:2
```

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/659.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
